app/integration/component-proxy/src/main/java/io/syndesis/integration/component/proxy/ComponentDefinition.java

```diff
@@ -77,6 +78,22 @@
         Optional<String> getFirstVersion();
     }
 
+    @Value.Immutable
+    @Value.Style(builder = "new")
+    @JsonDeserialize(builder = ImmutableRelationWhen.Builder.class)
+    interface RelationWhen {
+        String getId();
+        String getValue();
+    }
+
+    @Value.Immutable
+    @Value.Style(builder = "new")
+    @JsonDeserialize(builder = ImmutableRelation.Builder.class)
+    interface Relation {
```

Comment:
What's the difference between a `Relation` and a `PropertyRelation` ?

Reply:
Good question. We basically have duplicate models for `ComponentDefinition` and `ConnectorDescriptor`. I could rename `Relation` in Component definition to `PropertyRelation` to make it similar.